src/test/java/org/takes/http/FtSecureTest.java

```diff
@@ -0,0 +1,193 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.http;
+
+import com.jcabi.http.request.JdkRequest;
+import com.jcabi.http.response.RestResponse;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.ServerSocket;
+import java.net.URI;
+import javax.net.ssl.SSLServerSocketFactory;
+import org.apache.commons.io.IOUtils;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.takes.Request;
+import org.takes.Response;
+import org.takes.Take;
+import org.takes.facets.fork.FkRegex;
+import org.takes.facets.fork.TkFork;
+import org.takes.rq.RqLengthAware;
+import org.takes.rq.RqMethod;
+import org.takes.rq.RqPrint;
+import org.takes.rs.RsText;
+import org.takes.tk.TkFailure;
+
+/**
+ * Test case for {@link FtSecure}.
+ * @author Dragan Bozanovic (bozanovicdr@gmail.com)
+ * @version $Id$
+ * @since 0.25
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+@SuppressWarnings("PMD.TooManyMethods")
+public final class FtSecureTest {
+
+    /**
+     * Sets the certificate related properties.
+     */
+    @Before
+    public void before() {
+        final String file = this.getClass()
+            .getResource("/org/takes/http/keystore").getFile();
+        final String password = "abc123";
+        System.setProperty("javax.net.ssl.keyStore", file);
+        System.setProperty("javax.net.ssl.keyStorePassword", password);
+        System.setProperty("javax.net.ssl.trustStore", file);
+        System.setProperty("javax.net.ssl.trustStorePassword", password);
+    }
+
+    /**
+     * FtSecure can work.
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void justWorks() throws Exception {
+        secure(new TkFork(new FkRegex("/", "hello, world!"))).exec(
+            new FtRemote.Script() {
+                @Override
+                public void exec(final URI home) throws IOException {
+                    new JdkRequest(home)
+                        .fetch()
+                        .as(RestResponse.class)
+                        .assertStatus(HttpURLConnection.HTTP_OK)
+                        .assertBody(Matchers.startsWith("hello"));
+                }
+            }
+        );
+    }
+
+    /**
+     * FtSecure can work with a broken back.
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void gracefullyHandlesBrokenBack() throws Exception {
+        secure(new TkFailure("Jeffrey Lebowski")).exec(
```

Comment:
@pinaf Could you please elaborate on this? A link to the relevant docs or project conventions would be sufficient.